File: mobile_app/lib/config/api_config.dart
class ApiConfig {
  static const String baseUrl = 'http://172.20.10.11:8000/api';

  static const String register = '$baseUrl/accounts/register/';
  static const String login = '$baseUrl/accounts/login/';
  static const String me = '$baseUrl/accounts/me/';
  static const String restaurants = '$baseUrl/restaurants/';
  static const String menuItems = '$baseUrl/menu-items/';
  static const String orders = '$baseUrl/orders/';
  static const String reservations = '$baseUrl/reservations/';
  static const String registerDevice = '$baseUrl/notifications/register-device/';
  static const String unregisterDevice = '$baseUrl/notifications/unregister-device/';

  // Phone OTP (registration verification only)
  static const String requestOtp = '$baseUrl/accounts/request-otp/';
  static const String verifyOtp = '$baseUrl/accounts/verify-otp/';

  // Registration and default login
  static const String registerWithDetails = '$baseUrl/accounts/register-with-details/';
  static const String loginWithPassword = '$baseUrl/accounts/login-with-password/';

  // Optional PIN login (opt-in from Profile/Settings)
  static const String enablePin = '$baseUrl/accounts/enable-pin/';
  static const String disablePin = '$baseUrl/accounts/disable-pin/';
  static const String loginWithPin = '$baseUrl/accounts/login-with-pin/';

  static String imageUrl(String? path) {
    if (path == null || path.isEmpty) return '';
    if (path.startsWith('http')) return path;
    final base = baseUrl.replaceAll('/api', '');
    return '$base$path';
  }
}

File: admin-web/src/config/apiConfig.js
const BASE_URL = 'http://127.0.0.1:8000/api';

export const API_CONFIG = {
  baseURL: BASE_URL,
  login: `${BASE_URL}/accounts/login/`,
  register: `${BASE_URL}/accounts/register/`,
  me: `${BASE_URL}/accounts/me/`,
  restaurants: `${BASE_URL}/restaurants/`,
  myRestaurant: `${BASE_URL}/restaurants/mine/`,
  categories: `${BASE_URL}/categories/`,
  menuItems: `${BASE_URL}/menu-items/`,
  menuItemVariants: `${BASE_URL}/menu-item-variants/`,
  orders: `${BASE_URL}/orders/`,
  tableSessions: `${BASE_URL}/table-sessions/`,
  reservations: `${BASE_URL}/reservations/`,
};

export function imageUrl(path) {
  if (!path) return null;
  if (path.startsWith('http')) return path;
  const base = API_CONFIG.baseURL.replace('/api', '');
  return `${base}${path}`;
}

export default API_CONFIG;